ArduPlane/Parameters.cpp

```diff
@@ -1325,3 +1314,114 @@ void Plane::load_parameters(void)
+        switch (vtail_output) {
+        case MIXING_UPUP:
+        case MIXING_UPUP_SWP:
+            chan2->reversed_set_and_save_ifchanged(false);
+            chan4->reversed_set_and_save_ifchanged(false);
+            break;
+        case MIXING_UPDN:
+        case MIXING_UPDN_SWP:
+            chan2->reversed_set_and_save_ifchanged(false);
+            chan4->reversed_set_and_save_ifchanged(true);
+            break;
+        case MIXING_DNUP:
+        case MIXING_DNUP_SWP:
+            chan2->reversed_set_and_save_ifchanged(true);
+            chan4->reversed_set_and_save_ifchanged(false);
+            break;
+        case MIXING_DNDN:
+        case MIXING_DNDN_SWP:
+            chan2->reversed_set_and_save_ifchanged(true);
+            chan4->reversed_set_and_save_ifchanged(true);
+            break;
+        }
+        if (vtail_output < MIXING_UPUP_SWP) {
+            chan2->function_set_and_save(SRV_Channel::k_vtail_right);
+            chan4->function_set_and_save(SRV_Channel::k_vtail_left);
+        } else {
+            chan2->function_set_and_save(SRV_Channel::k_vtail_left);
+            chan4->function_set_and_save(SRV_Channel::k_vtail_right);
+        }
+    } else if (AP_Param::find_old_parameter(&elevon_info, &elevon_output) &&
+        elevon_output.get() != 0 &&
+        chan1->get_function() == SRV_Channel::k_aileron &&
+        chan2->get_function() == SRV_Channel::k_elevator &&
+        !chan1->function_configured() &&
+        !chan2->function_configured()) {
+        cliSerial->printf("convert elevon_output %u\n", elevon_output.get());
+        switch (elevon_output) {
+        case MIXING_UPUP:
+        case MIXING_UPUP_SWP:
+            chan2->reversed_set_and_save_ifchanged(false);
+            chan1->reversed_set_and_save_ifchanged(false);
+            break;
+        case MIXING_UPDN:
+        case MIXING_UPDN_SWP:
+            chan2->reversed_set_and_save_ifchanged(false);
+            chan1->reversed_set_and_save_ifchanged(true);
+            break;
+        case MIXING_DNUP:
+        case MIXING_DNUP_SWP:
+            chan2->reversed_set_and_save_ifchanged(true);
+            chan1->reversed_set_and_save_ifchanged(false);
+            break;
+        case MIXING_DNDN:
+        case MIXING_DNDN_SWP:
+            chan2->reversed_set_and_save_ifchanged(true);
+            chan1->reversed_set_and_save_ifchanged(true);
+            break;
+        }
+        if (elevon_output < MIXING_UPUP_SWP) {
+            chan1->function_set_and_save(SRV_Channel::k_elevon_right);
+            chan2->function_set_and_save(SRV_Channel::k_elevon_left);
+        } else {
+            chan1->function_set_and_save(SRV_Channel::k_elevon_left);
+            chan2->function_set_and_save(SRV_Channel::k_elevon_right);
+        }
+    }
+
+    // convert any k_aileron_with_input to aileron and k_elevator_with_input to k_elevator
+    for (uint8_t i=0; i<NUM_SERVO_CHANNELS; i++) {
```

Comment:
Discussed on mumble, I'm happy to let go of the _with_input item's if we can add a bitmask of channels that when overrides are active (and maybe/probably?) manual that the input channel is passed directly to the output channel. This is needed for some of the mixing I've played with in the past.